### PR TITLE
Add Github Login via Socialite

### LINKED ISSUE #388: Swap Gravatar for Github

### PR DESCRIPTION
**Pull request by TammyTee**
## Summary

This PR adds Laravel Socialite to authenticate users via GitHub. This will help against spam signups and allow us to use GitHub Avatars for new users instead of Gravatar images (closes #388). I kept in Gravatar as a fallback for legacy users.

This PR also removes the temp Honeypot setup and `RegisterController`.

**Todo Before Merge:**
- [x] Create an OAuth GitHub app for prod
- [x] Thoroughly test (app credentials in [Slack](https://tgtn.slack.com/archives/CGHUKLYH3/p1663010047839859)):
  - [x] You can sign up using GitHub and GitHub avatar is used
    - [x] On first sign up, you're redirected to the wizard, otherwise, the module index page
  - [x] You can sign in with a legacy user and Gravatar image is used
  - [x] Cancelling auth redirects back to onramp

### Screenshot(s)

| Updated Login Form |
| - |
| image |

| Gravatar message hidden on profile for Github users |
| - |
| image | 
| image |

| Changed Login/Register buttons to single "Get Started" cta |
| - |
| image |

### CODE REVIEW

**Review by MarcyLina — APPROVED** [review]
🤠

**Comment by TammyTee** [comment]
Revisiting to ensure the app returns gracefully after user cancels auth via GitHub

**Comment by TammyTee** [comment]
@darkboywonder when there's time, would you mind giving this a second review and test that you can do everything listed under "Todo Before Merge"? Thank you 🙏🏾

File: .env.example
Comment (darkboywonder) [inline .env.example:58]:
```suggestion

```

**Review by darkboywonder — APPROVED** [review]
🔥

**Pull request merged**
